_applications/flask_sparkpi.md

```diff
@@ -0,0 +1,90 @@
+---
+title: Flask SparkPi
+link: flask_sparkpi
+weight: 100
+layout: application
+menu_template: menu_tutorial_application.html
+description: |-
+  This source-to-image Python application combines the Apache Spark Pi
+  estimation example with the popular Flask Microframework. It provides an
+  HTTP microservice which will calculate the value of Pi on demand.
+project_links:
+- https://github.com/zmhassan/flask-pyspark-pi-example
+---
+
+<h1 id="introduction">Introduction</h1>
+
+The flask-sparkpi application provides an example of how to combine the
+power of Apache Spark with the flexibility of the
+[Flask microframework](http://flask.pocoo.org) to create an HTTP microservice
+that will deliver an approximation of [Pi](https://en.wikipedia.org/wiki/Pi)
+on request. The core of the processing for this application is based directly
+on the
+[Python Pi Estimation example](https://spark.apache.org/examples.html)
+([source](https://github.com/apache/spark/blob/master/examples/src/main/python/pi.py))
+from the upstream Apache Spark project.
+
+<h1 id="architecture">Architecture</h1>
+
+This application's architecture is quite straightforward. The source code
+repository is constructed to build a
+[source-to-image](https://github.com/openshift/source-to-image)
+deployment for OpenShift. Once built, OpenShift will deploy the driver
+application pod and then it will automatically deploy an Apache Spark cluster
+of one master pod and one worker pod. Finally, the driver and the cluster pods
+will be connected, at which point the service will be ready for access.
+
+<h1 id="installation">Installation</h1>
+
+Installation of this application is quite simple using the `oc` command line
+tool. Once you are logged in to your OpenShift project, follow these
+instructions:
+
+
+1. Follow the [Get Started](/get-started) guide to ensure that you are
+   connected to an OpenShift project with Oshinko installed
+
+1. Launch spring-sparkpi
+
+   ```bash
+   oc new-app --template oshinko-pyspark-build-dc  \
+       -p APPLICATION_NAME=flask-sparkpi2 \
+       -p GIT_URI=https://github.com/zmhassan/flask-pyspark-pi-example.git  \
+       -p APP_FILE=app.py
+   ```
+
+1. Expose an external route
+
+   ```bash
+   oc expose svc/flask-sparkpi2
+   ```
+
+<h1 id="usage">Usage</h1>
+
+With the route exposed to your service, you can now begin making queries to
+approximate the value of Pi. This can be done with a web browser or other
+tool capable of reading HTTP, such as `curl`.
+
+**For example**
+
+```bash
+$ curl http://`oc get routes/flask-sparkpi2 --template='{{.spec.host}}'`
```

Comment:
the liquid template framework that is used by jekyll is giving a warning on this line, it needs to have the raw indicators around the `{{.spec.host}}` part because liquid thinks it's a variable. something like this should work:

```
$ curl http://`oc get routes/flask-sparkpi2 --tempalte='{% raw %}{{.spec.host}}{% endraw %}'`
```